### ISSUE TITLE
Eslint parsing failure

### ISSUE THREAD

**Issue by nashaofu** [babel/babel-eslint#599]
Eslint configuration:
```js
// https://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint'
  },
  env: {
    browser: true,
  },
  extends: [
    // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
    // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
    'plugin:vue/essential',
    // https://github.com/standard/standard/blob/master/docs/RULES-en.md
    'standard'
  ],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  // add your custom rules here
  rules: {
    'no-return-await': 'off',
    // allow async-await
    'generator-star-spacing': 'off',
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off'
  }
}
```

Output:
```js
  ✘  http://eslint.org/docs/rules/  Parsing error: await is a reserved word

  19 |    */
  20 |   if (process.platform === 'win32') {
> 21 |     return await Promise.all(displays.map(async (display, index) => {
     |            ^
  22 |       return await new Promise((resolve, reject) => {
  23 |         desktopCapturer.getSources({
  24 |           types: ['screen'],
  src\renderer\assets\js\getSources.js:21:12
      return await Promise.all(displays.map(async (display, index) => {
              ^


✘ 1 problem (1 error, 0 warnings)
```

And source code:
```js
import { desktopCapturer } from 'electron'

export default async (displays, bounds) => {
  // 图片定位要以窗口最左侧开始
  // 但在窗口可能不是在(0, 0)位置
  let dx = bounds.x
  let dy = bounds.y
  displays.forEach(display => {
    if (dx > display.x) {
      dx = display.x
    }
    if (dy > display.y) {
      dy = display.y
    }
  })
  /**
   * 每一个屏幕截一张图是为了让每一张图片都尽可能的清晰
   * 图片经过缩放之后质量损失非常大
   */
  if (process.platform === 'win32') {
    return await Promise.all(displays.map(async (display, index) => {
      return await new Promise((resolve, reject) => {
        desktopCapturer.getSources({
          types: ['screen'],
          thumbnailSize: {
            width: display.width,
            height: display.width
          }
        }, (error, sources) => {
          if (error) {
            return reject(error)
          }
          resolve({
            x: display.x - dx,
            y: display.y - dy,
            width: display.width,
            height: display.height,
            thumbnail: sources[index].thumbnail
          })
        })
      })
    }))
  } else {
    return await new Promise((resolve, reject) => {
      desktopCapturer.getSources({
        types: ['screen'],
        thumbnailSize: {
          width: bounds.width,
          height: bounds.width
        }
      }, (error, sources) => {
        if (error) {
          return reject(error)
        }
        resolve([{
          x: 0,
          y: 0,
          width: bounds.width,
          height: bounds.height,
          thumbnail: sources[0].thumbnail
        }])
      })
    })
  }
}
```

And I can run success after I remove babel-eslint option, the new configuration:

```js
// https://eslint.org/docs/user-guide/configuring

module.exports = {
  root: true,
//   parserOptions: {
//     parser: 'babel-eslint'
//   },
  env: {
    browser: true,
  },
  extends: [
    // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
    // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
    'plugin:vue/essential',
    // https://github.com/standard/standard/blob/master/docs/RULES-en.md
    'standard'
  ],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  // add your custom rules here
  rules: {
    'no-return-await': 'off',
    // allow async-await
    'generator-star-spacing': 'off',
    // allow debugger during development
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off'
  }
}
```

**Comment by loganfsmyth** [comment]
What version of `babel-eslint` are you using? I also believe `parser` is a top-level option, e.g.

```
root: true,
parser: 'babel-eslint',
```

**Comment by nashaofu** [comment]
the version of `babel-eslint` is 8.2.1, And I modify the configuration file as well as yours, but it still didn't work.

**Comment by chrisdothtml** [comment]
@loganfsmyth moving `parser` to the top level fixed it for me 🎉

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
